webapp/src/EventListener/NoSessionCookieForApiListener.php

```diff
@@ -0,0 +1,24 @@
+<?php declare(strict_types=1);
+
+namespace App\EventListener;
+
+use Symfony\Component\EventDispatcher\Attribute\AsEventListener;
+use Symfony\Component\HttpKernel\Event\ResponseEvent;
+
+// The AbstractSessionListener (which sets the cookie) has a priority of -1000, so we need to
+// set a priority of -1001 to run before it.
+#[AsEventListener(priority: -1001)]
+class NoSessionCookieForApiListener
+{
+    public function __invoke(ResponseEvent $event): void
+    {
+        // We do not want to set the session cookie for API requests. Since the firewall is
```

Comment:
Would it be hard to use the session for the judgehosts/libcurl (and would it give us an advantage)?

Reply:
We would need a way to *DO* keep the session cookie, then use it once to log in and then keep passing the session token.

I don't know how big the advantage is, it means we don't have to check the (encrypted) password for every request. If we want this, I'd say we do it in a separate PR.